pkg/skaffold/schema/profiles.go

```diff
@@ -309,44 +329,76 @@ func overlayStructField(config interface{}, profile interface{}) interface{} {
 
 	for i := 0; i < profileValue.NumField(); i++ {
 		fieldType := t.Field(i)
-		overlay := overlayProfileField(yamltags.YamlName(fieldType), configValue.Field(i).Interface(), profileValue.Field(i).Interface())
+		yamlFieldName := yamltags.YamlName(fieldType)
+		var first rune
+		for _, c := range yamlFieldName {
+			first = c
+			break
+		}
+		if !unicode.IsLower(first) {
+			yamlFieldName = "-"
+		}
+		overlay := overlayProfileField(profileName, yamltags.YamlName(fieldType), yamlFieldName, fieldPath, fieldPathElementsOverrodeByProfile, configValue.Field(i).Interface(), profileValue.Field(i).Interface())
 		finalConfig.Elem().FieldByName(fieldType.Name).Set(reflect.ValueOf(overlay))
 	}
 	return reflect.Indirect(finalConfig).Interface() // since finalConfig is a pointer, dereference it
 }
 
-func overlayProfileField(fieldName string, config interface{}, profile interface{}) interface{} {
+// I could either get struct names in a flat manner and just not care about name collisions for now
+// OR I could make a tree of fieldNames and then check the tree when doing the profile yaml node stuff
+func overlayProfileField(profileName, fieldName string, yamlFieldName string, fieldPath []string, fieldPathElementsOverrodeByProfile map[string]configlocations.YAMLOverrideInfo, config interface{}, profile interface{}) interface{} {
 	v := reflect.ValueOf(profile) // the profile itself
 	t := reflect.TypeOf(profile)  // the type of the profile, used for getting struct field types
 	log.Entry(context.TODO()).Debugf("overlaying profile on config for field %s", fieldName)
+
+	if yamlFieldName != "-" {
+		fieldPath = append(fieldPath, yamlFieldName)
+	}
+
 	switch v.Kind() {
 	case reflect.Struct:
 		// check the first field of the struct for a oneOf yamltag.
 		if util.IsOneOfField(t.Field(0)) {
-			return overlayOneOfField(config, profile)
+			return overlayOneOfField(profileName, fieldPath, fieldPathElementsOverrodeByProfile, config, profile)
 		}
-		return overlayStructField(config, profile)
+		return overlayStructField(profileName, fieldPath, fieldPathElementsOverrodeByProfile, config, profile)
 	case reflect.Slice:
 		// either return the values provided in the profile, or the original values if none were provided.
 		if v.Len() == 0 {
 			return config
 		}
+		fieldPathElementsOverrodeByProfile["/"+strings.Join(fieldPath, "/")] = configlocations.YAMLOverrideInfo{
```

Comment:
I think it might be easier/cleaner to replace these `strings.Join()` calls with [`path.Join()`](https://pkg.go.dev/path#Join) since it automatically joins with `/`

Reply:
Done